### ISSUE TITLE
Automatically turn the chessboard around to your position when you're analyzing the chessboard

### ISSUE THREAD

**Issue by olivertzeng** [lichess-org/mobile#837]
iOS 16.5
iPhone 14
Lichess Beta 0.8.18!![image](https://github.com/lichess-org/mobile/assets/86348833/87f0ce4c-fad2-4a36-a94a-43c02b8aedea)![image](https://github.com/lichess-org/mobile/assets/86348833/220e30aa-9737-47c2-8a01-7e11c899284a)

**Comment by BAW2501** [comment]
Could you provide some clarification? It seems like you are importing a PGN and want it to rotate based on the player's name from the PGN. Is that correct? Also, please outline the steps to reproduce this problem.

**Comment by olivertzeng** [comment]
> Could you provide some clarification? It seems like you are importing a PGN and want it to rotate based on the player's name from the PGN. Is that correct? Also, please outline the steps to reproduce this problem.

sorry for not providing that info. steps to reproduce:
set if your username is foo and you were black, and another username called bar who played white, and you imported that game's PGN into Lichess it'll always show the white pieces on "your side" instead of what you were playing.
what i expect:
automatically detect your username and turn the board accordingly

Thanks @BAW2501 for responding!

**Comment by ijm8710** [comment]
This is a good point. I also agree it would be helpful for analysis if this was adjusted for it to auto-adjust based on what color you are playing if it is defined.

**Comment by olivertzeng** [comment]
> not sure if this would be the desired behavior of the PGN importer as there is no guarantee that user foo in the PGN is foo from Lichess, I'll wait on the input of the main dev before I try my hand at it.

Well the example game I've imported Lichess was played in Lichess as well.

**Comment by Mauritz8** [comment]
> Well the example game I've imported Lichess was played in Lichess as well.

I'm a bit confused, are you importing lichess games into lichess? Are they not already in lichess then?

**Comment by olivertzeng** [comment]
> > Well the example game I've imported Lichess was played in Lichess as well.
> 
> 
> 
> I'm a bit confused, are you importing lichess games into lichess? Are they not already in lichess then?

Yes
It was this game
![image](https://github.com/user-attachments/assets/c78dfb10-f705-490b-8714-ae3a4a46d13e)
But with less detailed suggestions compared to imported games.
![image](https://github.com/user-attachments/assets/57f875c0-924f-484b-aee5-8fb4461a85a9)
So sometimes I import game from exported Lichess games into lichess

**Comment by Mauritz8** [comment]
Interesting workaround. So, what you really want is the comments? If that's the case I think we should address that, rather than adding features for the workaround.

I also wonder, how do you even open imported games in the app?

**Comment by olivertzeng** [comment]
No it's kinda like a separate issue though

And below this is how I import a game
https://github.com/user-attachments/assets/e922c1f0-0004-4522-8127-1506b97b6ff4

**Comment by olivertzeng** [comment]
> I also wonder, how do you even open imported games in the app?

It's not directly imported though it's exported then imported

**Comment by Mauritz8** [comment]
> No it's kinda like a separate issue though

No? So, why don't you just view the game from your profile then?

**Comment by olivertzeng** [comment]
> > No it's kinda like a separate issue though
> 
> No? So, why don't you just view the game from your profile then?

It’s an issue I would like to submit later for now but it’s still an issue that makes me export and then import the game in lichess after some games

**Comment by veloce** [comment]
So the whole thing is about seeing the all variations in an analysed game?

We can add an action to toggle the display of all variations at once (currently it's only possible with a context action for each variation, not the most convenient).

Also I don't like to add too much settings, but we could have another analysis settings to decide should be shown from the start or not.

**Comment by olivertzeng** [comment]
[**@veloce**](https://github.com/veloce) commented on [Jul 16, 2024, 7:29 AM GMT+8](https://github.com/lichess-org/mobile/issues/837#issuecomment-2229614753 "2024-07-15T23:29:08Z - Replied by Github Reply Comments"):
> So the whole thing is about seeing the all variations in an analysed game?
> 
> We can add an action to toggle the display of all variations at once (currently it's only possible with a context action for each variation, not the most convenient).
> 
> Also I don't like to add too much settings, but we could have another analysis settings to decide should be shown from the start or not.

no

i meant automatically facing black pieces to the player if the player's username matches the Lichess's username

**Comment by veloce** [comment]
The import PGN feature is not really meant to import lichess games in the first place...

**Comment by olivertzeng** [comment]
> The import PGN feature is not really meant to import lichess games in the first place...

It's because that imported games have more info than just viewing from the profile. Please view all of the earlier comments in this thread

**Comment by olivertzeng** [comment]
> > > Well the example game I've imported Lichess was played in Lichess as well.
> > 
> > 
> > 
> > I'm a bit confused, are you importing lichess games into lichess? Are they not already in lichess then?
> 
> Yes
> It was this game
> ![image](https://github.com/user-attachments/assets/3a71e2a4-2217-443e-88eb-53cc234cc402)
> But with less detailed suggestions compared to imported games.(something like (blunder xx was the best move))
> ![image](https://github.com/user-attachments/assets/57f875c0-924f-484b-aee5-8fb4461a85a9)
> So sometimes I import game from exported Lichess games into lichess
> 
> Sorry for the confusion 

This comment is what I meant

**Comment by veloce** [comment]
Have you read my comment: https://github.com/lichess-org/mobile/issues/837#issuecomment-2229614753 ?

You want to see the comments in italics and the variations (between parenthesis), I got it.

**Comment by olivertzeng** [comment]
> Have you read my comment: https://github.com/lichess-org/mobile/issues/837#issuecomment-2229614753 ?
> 
> 
> 
> You want to see the comments in italics and the variations (between parenthesis), I got it.

Sorry for being mean
But I got the point

**Comment by bhavikpanchal1111** [comment]
I am willing to solve this issue, so I would need developer access.

**Comment by olivertzeng** [comment]
> I am willing to solve this issue, so I would need developer access.

You can fork this repository and make nessesary fixes and changes. After that, try to file a pull request to them. They will then decide if it's good or not.

**Comment by veloce** [comment]
Closing this as I think the main issue reported here is fixed by #1053

**Comment by olivertzeng** [comment]
> Closing this as I think the main issue reported here is fixed by #1053

I've reviewed the pr, but cannot find as of any sign of the fixes for this issue. Could you be so kind to show me which part of the pr includes that fix? thank you!

**Comment by olivertzeng** [comment]
> Closing this as I think the main issue reported here is fixed by #1053 

After testing the latest version of lichess beta I find this issue unfixed though

I exported the PGN from lichess as me playing black and the board doesn't automatically turn around like I normally do see if I click the game instead of exporting

But to see if the board needs a "turnaround" you could just try to  see if the lichess username that's logged in is the same as the player in black in the pgn

**Comment by veloce** [comment]
Well importing PGN in analysis is meant for external PGN sources, not the lichess PGN. While you can do it, we won't add logic to it. I thought the original issue which made you import lichess PGN in the app was kind of solved now, thus closing the issue.

**Comment by olivertzeng** [comment]
> Well importing PGN in analysis is meant for external PGN sources, not the lichess PGN. While you can do it, we won't add logic to it. I thought the original issue which made you import lichess PGN in the app was kind of solved now, thus closing the issue. 

My original issue is actually not that

My issue doesn't matter where your pgn import from, it's about the part where there is `[Black: "whatever_username"]` which I believe that this tag is universal through different chess websites that supports PGN exporting.

For example if my lichess username matches the username that's in the `Black` tag, turn the board around automatically. It's actually useful when I want to import my game from chess.com
